### PR TITLE
fix: fix languages design on desktop

### PR DESCRIPTION
**Pull request by PedroMiguelSS**
This PR updates desktop languages bar to the new design. 
Major changes:

- available locales array on `intl` config file has changed: it was an array of strings (locales acronyms) and it is an object now containing `acronym` and `fullForm`. 
E.g.: 
```javascript
availableLocales: [
        { acronym: 'en', fullForm: 'English' },
        { acronym: 'pt', fullForm: 'Português (PT)' }
    ]
```
- new `utils` file that exports 3 functions:

1. `getLocalesAcronym` - returns an array containing only the acronyms;
2. `getLocalesFullForm` - returns an array containing only the locales full form;
3. `getIndexByAcronym` - returns the index of a given acronym.

### CODE REVIEW

File: src/shared/components/header/index.js [hunk 1]
```diff
@@ -1,18 +1,63 @@
...
+    constructor() {
+        super();
+
+        this.state = {
+            scrolled: false,
+        };
+    }
+
+    componentDidMount() {
+        window.addEventListener('scroll', this.handleScroll);
```
Comment (vasco-santos) [inline src/shared/components/header/index.js]: Did you consider using [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)? 

You will be listening to the scroll events all the time this way,

File: src/shared/components/header/index.js [hunk 2]
```diff
@@ -1,18 +1,63 @@
...
+                {/* <Banner /> */}
+                <LocalesBar scrolled={ scrolled } ref={ this.handleLocalesBarRef } />
+                <DesktopNavbar scrolled={ scrolled } localesBarHeight={ localesBarHeight } />
+                <MobileNavbar />
+            </header>
+        );
+    }
+
+    handleScroll = () => {
+        const scrollY = window.scrollY;
```
Comment (vasco-santos) [inline src/shared/components/header/index.js]: As you use it several times:
```js
const { scrolled } = this.state
```

File: src/shared/components/locales-dropdown/index.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React, { Component } from 'react';
 import { injectIntl } from 'react-intl';
```
Comment (vasco-santos) [inline src/shared/components/locales-dropdown/index.js]: Should the name be different than dropdown?

Comment (vasco-santos) [reply src/shared/components/locales-dropdown/index.js]: oh wait, this probably is for mobile, right?

File: src/shared/components/locales-dropdown/index.module.css [hunk 4]
```diff
@@ -88,5 +88,7 @@
 
     & .show {
         display: block;
+
+        /* height: 100px; */
```
Comment (vasco-santos) [inline src/shared/components/locales-dropdown/index.module.css]: remove this?

File: src/shared/components/locales-dropdown/index.module.css [hunk 5]
```diff
@@ -65,7 +61,11 @@
         z-index: 1;
         overflow: auto;
         display: none;
+
+        /* height: 0; */
```
Comment (vasco-santos) [inline src/shared/components/locales-dropdown/index.module.css]: remove this?

File: src/shared/components/locales-dropdown/index.module.css (hunk 4 above)
Comment (PedroMiguelSS) [reply src/shared/components/locales-dropdown/index.module.css]: I will update the animation of this dropdown in the future. And I'll probably need this `height` to animate it.

File: src/shared/components/locales-dropdown/index.module.css (hunk 5 above)
Comment (PedroMiguelSS) [reply src/shared/components/locales-dropdown/index.module.css]: Same thing here.

File: src/shared/components/locales-dropdown/index.js (hunk 3 above)
Comment (PedroMiguelSS) [reply src/shared/components/locales-dropdown/index.js]: Yes, you're right. This dropdown will be used on mobile devices.

File: src/shared/components/header/index.js (hunk 1 above)
Comment (PedroMiguelSS) [reply src/shared/components/header/index.js]: I didn't consider IntersectionObserver. But I'll create a card to implement it later.

File: src/shared/components/navbar/desktop/index.module.css [hunk 6]
```diff
@@ -37,7 +37,7 @@
 }
 
 .default {
-    padding: 60px 0 50px 0;
+    padding: 35px 0 35px 0;
```
Comment (fsdiogo) [inline src/shared/components/navbar/desktop/index.module.css]: Use shorthand property.

**Pull request merged**